core/services/src/main/java/org/exoplatform/services/cms/documents/impl/DocumentServiceImpl.java

```diff
@@ -501,21 +506,25 @@ private String getSharedLink(Node currentNode, Node rootNode) {
         currentNode = linkManager.getTarget(currentNode);
       }
       List<String> path = new ArrayList<>();
-      Node targetNode = null;
       boolean existingSymlink = false;
       for (NodeIterator it = shared.getNodes(); it.hasNext(); ) {
         Node node = it.nextNode();
         path.add(((NodeImpl) node).getInternalPath().getAsString());
         if (path.contains(((NodeImpl) shared).getInternalPath().getAsString() + "[]" + currentNode.getName() + ":1")) {
           existingSymlink = true;
-          targetNode = node;
+          link = node;
           break;
         }
       }
-      if (existingSymlink) {
-        link = targetNode;
-      } else {
-        link = linkManager.createLink(shared, currentNode);
+      if (!existingSymlink) {
+        for (NodeIterator it = ActivityNode.getNodes(); it.hasNext(); ) {
+          Node node = it.nextNode();
+          path.add(((NodeImpl) node).getInternalPath().getAsString());
+          if (path.contains(((NodeImpl) ActivityNode).getInternalPath().getAsString() + "[]" + currentNode.getName() + ":1")) {
```

Comment:
What are you trying to check with this test condition ?

Reply:
If the file is not existing in the shared folder (symlink) so that means its' not a shared file, i could easily say if (link == null) but i let it as it is to be more expressive.

Reply:
If the file (node) was found in the ActivityStreamFolder it will be in this form for example ([]:1[]Groups:1[]spaces:1[]testspace:1[]Documents:1[]Activity Stream Documents:1[]smtp_conf:1;)
so it will assign it to the link variable and break the loop execution

Reply:
There are other cleaner ways to do it, let's have a call and I should be able to propose a better way to doit.

Reply:
> There are other cleaner ways to do it, let's have a call and I should be able to propose a better way to doit.

Okay